src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -576,3 +611,513 @@ void JasmineGraphFrontEnd::removeGraph(std::string graphID, void *dummyPt) {
+
+long JasmineGraphFrontEnd::countGlobalTriangles(int graphId, void *dummyPt) {
+    long result;
+    long startVid;
+    long endVid;
+    int centralPartitionCount = 0;
+    std::string query = "select CENTRALPARTITIONCOUNT from ACACIA_META.GRAPH where IDGRAPH=" + std::to_string(graphId);
+
+    SQLiteDBInterface *sqlite = (SQLiteDBInterface *) dummyPt;
+    std::vector<vector<pair<string, string>>> results = sqlite->runSelect(query);
+
+    map<long, unordered_set<long>> localSubgraphMap;
+
+    if (results.size() > 0) {
+        vector<pair<string, string>> resultPairVector = results.at(0);
+        if (resultPairVector.size() >0) {
+            std::string resultString = resultPairVector.at(0).second;
+            centralPartitionCount = atoi(resultString.c_str());
+        }
+    }
+
+    for (int i=0; i< centralPartitionCount; i++) {
+        JasmineGraphHashMapCentralStore *centralStore = new JasmineGraphHashMapCentralStore(graphId, i);
+        centralStore->loadGraph();
+
+        map<long, unordered_set<long>> edgeList2 = centralStore->getUnderlyingHashMap();
+        map<long, unordered_set<long>>::iterator edgeListIterator;
+        long firstVertex = 0;
+
+        for (edgeListIterator = edgeList2.begin(); edgeListIterator != edgeList2.end(); ++edgeListIterator) {
+            firstVertex = edgeListIterator->first;
+            unordered_set<long> hs = localSubgraphMap[firstVertex];
+
+            unordered_set<long> hs2 = edgeListIterator->second;
+
+            for (unordered_set<long>::iterator it = hs2.begin(); it != hs2.end(); ++it) {
+                hs.insert(*it);
+            }
+
+            localSubgraphMap[firstVertex] = hs;
+        }
+
+    }
+
+    map<long, unordered_set<long>> degreeMap;
+    map<long,long> degreeReverseLookupMap;
+
+    long degree =0;
+    startVid =0;
+    map<long,unordered_set<long>>::iterator localSubgraphMapIterator;
+
+    for (localSubgraphMapIterator = localSubgraphMap.begin(); localSubgraphMapIterator != localSubgraphMap.end(); ++ localSubgraphMapIterator) {
+        startVid = localSubgraphMapIterator->first;
+        unordered_set<long> items = localSubgraphMapIterator->second;
+
+        degree = items.size();
+
+        unordered_set<long> degreeSet = degreeMap[degree];
+        degreeSet.insert(startVid);
+        degreeMap[degree] = degreeSet;
+    }
+
+    long triangleCount =0;
+    long v1 = 0;
+    long v2 = 0;
+    long v3 = 0;
+    map<long, map<long, std::vector<long>>> triangleTree;
+    vector<long> degreeListVisited;
+
+    map<long, unordered_set<long>>::iterator degreeMapIterator;
+
+    for (degreeMapIterator = degreeMap.begin(); degreeMapIterator != degreeMap.end(); ++ degreeMapIterator) {
+        long kkkk = degreeMapIterator->first;
+
+        unordered_set<long> vVertices = degreeMap[kkkk];
+
+        unordered_set<long>::iterator vVerticesIterator;
+
+        for (vVerticesIterator = vVertices.begin(); vVerticesIterator != vVertices.end(); ++vVerticesIterator) {
```

Comment:
We should move this triangle counting code to query/algorithms package. We should not have that logic duplicated here.